### PR TITLE
[erc721-multi-registry-weighted] Add FRWC token weighted strategy

### PR DESCRIPTION
**Pull request by Meph1587**
Add FRWC token weighted strategy.

Changes proposed in this pull request:
- Add a strategy to assign votes to holders of Forgotten Runes Wizard Cult tokens
- No strategy is present to allow multiple ERC721 to be counted with different weights
- Strategy Passes all tests, including if run with >500 addresses

### CODE REVIEW

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Hi I see this strategy is equal to `erc721-multi-registry` except weigts, so is it possible to rename it as `erc721-multi-registry-weighted` ? so others can reuse it if they want to

**Comment by Meph1587** [comment]
Updated that, can you review again? Thanks!

**Comment by cryppadotta** [comment]
Looks good to me 👍

File: src/strategies/erc721-multi-registry-weighted/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+
+export const author = 'dievardump';
```
Comment (ChaituVR) [inline src/strategies/erc721-multi-registry-weighted/index.ts:4]:
```suggestion
export const author = 'Meph1587';
```

File: src/strategies/erc721-multi-registry-weighted/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const calls: any[] = [];
+  const multipliers: any[] = [];
+  options.tokens.map((token, idx) => {
+    addresses.forEach((address: any) => {
+      calls.push([token, 'balanceOf', [address]]);
+      multipliers.push(options.weights[idx])
```
Comment (ChaituVR) [inline src/strategies/erc721-multi-registry-weighted/index.ts:26]:
```suggestion
      multipliers.push(options.weights[idx] || 1)
```

File: src/strategies/erc721-multi-registry-weighted/index.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const calls: any[] = [];
+  const multipliers: any[] = [];
+  options.tokens.map((token, idx) => {
+    addresses.forEach((address: any) => {
+      calls.push([token, 'balanceOf', [address]]);
+      multipliers.push(options.weights[idx] || 1)
```
Comment (ChaituVR) [inline src/strategies/erc721-multi-registry-weighted/index.ts:26]:
```suggestion
      multipliers.push(options.weights?[idx] || 1)
```

File: src/strategies/erc721-multi-registry-weighted/index.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const calls: any[] = [];
+  const multipliers: any[] = [];
+  options.tokens.map((token, idx) => {
+    addresses.forEach((address: any) => {
+      calls.push([token, 'balanceOf', [address]]);
+      multipliers.push(options.weights?[idx] || 1)
```
Comment (ChaituVR) [inline src/strategies/erc721-multi-registry-weighted/index.ts:26]:
```suggestion
      multipliers.push(options.weights[idx] || 1)
```

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**